jwst/resample/resample_step.py

```diff
@@ -176,16 +176,26 @@ def load_custom_wcs(asdf_wcs_file, output_shape=None):
 
         with asdf.open(asdf_wcs_file) as af:
             wcs = deepcopy(af.tree["wcs"])
-            wcs.pixel_area = af.tree.get("pixel_area", None)
-            wcs.array_shape = af.tree.get("pixel_shape", None)
-            wcs.array_shape = af.tree.get("array_shape", None)
+            pixel_area = af.tree.get("pixel_area", None)
+            pixel_shape = af.tree.get("pixel_shape", None)
+            array_shape = af.tree.get("array_shape", None)
+
+        if not hasattr(wcs, "pixel_area") or wcs.pixel_area is None:
+            wcs.pixel_area = pixel_area
+        if not hasattr(wcs, "pixel_shape") or wcs.pixel_shape is None:
+            wcs.pixel_shape = pixel_shape
+        if not hasattr(wcs, "array_shape") or wcs.array_shape is None:
+            wcs.array_shape = array_shape
 
         if output_shape is not None:
             wcs.array_shape = output_shape[::-1]
             wcs.pixel_shape = output_shape
         elif wcs.pixel_shape is not None:
             wcs.array_shape = wcs.pixel_shape[::-1]
-        elif wcs.array_shape is not None:
+        elif wcs.array_shape is not None:  # pragma: no cover
+            # This branch is not currently reachable under normal
```

Comment:
After testing with gwcs and asdf, I found that `array_shape` and `pixel_shape` are preserved in the saved asdf file, but `pixel_area` is not.

Reply:
Thanks for testing!  

I realized what happened here - that clause was not reached in my tests because array_shape is automatically set when pixel_shape is set.  This comment should be removed, and array_shape setting inside the wcs should be tested similarly to the way pixel_shape is tested in the larger unit tests.